File: JavaScript/Date.js

// 現在の日時
var date1 = new Date().toLocaleString({timeZone: 'Asia/Tokyo'})
console.log(date1)

// 日時文字列から生成
date1 = new Date('2021/01/07 07:30').toLocaleString({timeZone: 'Asia/Tokyo'})
console.log(date1)

// 年月日時分から生成（月は1を引く）
date1 = new Date(2021, 1-1, 7, 7, 30).toLocaleString({timeZone: 'Asia/Tokyo'})
console.log(date1)

// ミリ秒から生成
date1 = new Date(1617206523000)
console.log(date1)

// 現在のミリ秒を取得
console.log(new Date().getTime())

// 文字列からミリ秒へ
var millsec = Date.parse('2021-04-01T01:02:03+09:00')
console.log(millsec)
console.log(new Date(millsec))

// 文字列からDateへ
date1 = new Date('2021-04-01T01:02:03+09:00')
console.log(date1)

// Dateから文字列
date1 = new Date()
dateStr = date1.toISOString()
console.log(dateStr)

// 年月日時分秒を個別に設定
date1 = new Date()
date1.setFullYear(2022)
date1.setMonth(3)
date1.setDate(7)
date1.setHours(01)
date1.setMinutes(02)
date1.setSeconds(16)
date1.setMilliseconds(55)

// 年月日時分秒を個別に取得
year = date1.getFullYear()
month = date1.getMonth() + 1
date = date1.getDate()
day = date1.getDay() // 曜日（日曜日(0)～土曜日(6））
hour = date1.getHours()
min = date1.getMinutes()
sec = date1.getSeconds()
msec = date1.getMilliseconds()
console.log( year + '年' + month + '月' + date + '日' + hour + '時' + min + '分' + sec + '秒.' + msec)

// 日時の比較
date1 = new Date('2020/8/1')
date2 = new Date('2020/8/2')
console.log(date2 > date1) // true

File: Ruby/Date.rb
# ////////// Timeクラス //////////
# （通常は、DateTimeではなくTimeを利用すればOK）

require "time"

# 生成
tm = Time.new
p tm
tm = Time.now
p tm
tm = Time.local(2021, 1, 2, 3, 4, 5)
p tm
tm = Time.parse('2021-04-01T02:03:04')
p tm
tm = tm + 10 # 10秒後
p tm

# アクセス
tm = Time.new
p tm.year
p tm.month
p tm.day
p tm.wday
p tm.yday
p tm.zone
p tm.hour
p tm.min
p tm.sec

# 文字列 <-> Time
tmStr = tm.strftime('%Y/%m/%d %H:%M:%S') 
p tmStr
tm = Time.parse('2021-04-01T02:03:04')
p tm

# ISO8601
tmStr = Time.now.iso8601
p tmStr
tm = Time.parse(tmStr)
p tm

# 比較
require "active_support/time"
time = Time.now
puts time.yesterday < time # true



# ////////// DateTimeクラス //////////

require "date"

# 生成
dt = DateTime.new
p dt
dt = DateTime.new(2021, 4, 1, 2, 3, 4)
p dt
dt = DateTime.parse('2021-04-01T02:03:04')
p dt

# アクセス
p dt.year
p dt.month
p dt.day
p dt.wday
p dt.yday
p dt.zone
p dt.hour
p dt.min
p dt.sec

# 文字列 <-> DateTime
dtStr = dt.strftime('%Y/%m/%d %H:%M:%S') 
p dtStr
dt = DateTime.parse(dtStr)
p dt
